### DISCUSSION TITLE
Number sorting in AG-grid table

### DISCUSSION THREAD

**Discussion by davidnetten** [windmill-labs/windmill#4166] [Questions] [answered]
I'm using this to show a value in a table in an app

{
  "flex": 1,
  "sort": "desc",
  "field": "usage_data",
  "headerName": "Data Usage (MB)",
  "cellRendererType": "text",
  "filter": false,
  "valueFormatter": "(value/1024/1024).toFixed(2)"
}

The values in the table are correctly displayed, but when I sort descending, the number is sorter alphabetically, so 901 comes before 91. 

I tried to set cellRendererType to number as per the documentation, but that does not neem to work. The strange thing is that the calulated value is correct, just the sorting handles it as a string.

**Comment by davidnetten** [comment]
I learned that the output is still not explicitly a number, but also with parseInt(value / 1024 / 1024) it does not sort the table on the numeric value but the alphabetic value (descending: 310 comes before 3000)

I noticed that the dropdown only shows Type and Link, which might indicate the table simply does not accept numeric values, correct?

**Comment by davidnetten** [comment]
Adding "cellDataType": "number", also seems to fail...

**Comment by davidnetten** [comment, marked as answer]
For some reason, setting that to object (toggle raw, that add this line: "cellDataType": "object",) fixes is...

**Comment by a-coors** [comment]
`valueGetter: (params) => parseInt(params.data.id) ?? params.data.id,`
